Automattic/wp-calypso
# Framework: Use `he` package for string decoding in all environments

Currently, we split the implementation for `lib/formatting`'s `decodeEntities` by environment, using the [`he` package](https://www.npmjs.com/package/he) on the server and a `<textarea>`-based solution in the browser. Upon further investigation, it was found that including `he` in the browser [would not cause a significant size increase to the base bundle](https://www.diffchecker.com/npaqhvu7), and would provide a consistent decoding solution, eliminating the potential for differences between environments.

This pull request seeks to remove the browser-specific implementation of `decodeEntities`, instead importing the `he` package `decode` method in the base `lib/formatting` module.

__Testing instructions:__

Confirm newly introduced Mocha tests pass by running `make test` from the project root directory, or directly from `client/lib/formatting`.

Verify that sections making use of `decodeEntities` remain unaffected (e.g. Reader post titles).

/cc @blowery , @ehg
